contracts/drafts/ERC777/ERC777.sol

```diff
@@ -0,0 +1,402 @@
+        address to,
+        uint256 amount,
+        bytes calldata data,
+        bytes calldata operatorData
+    )
+    external
+    {
+        address holder;
+        if (from == address(0)) {
+            holder = msg.sender;
+        } else {
+            holder = from;
+            require(isOperatorFor(msg.sender, holder));
+        }
+
+        _send(msg.sender, holder, to, amount, data, operatorData);
+    }
+
+    /**
+    * @dev Burn the amount of tokens from the address msg.sender
+    * @param amount uint256 amount of tokens to transfer
+    * @param data bytes extra information provided by the token holder
+     */
+    function burn(uint256 amount, bytes calldata data) external {
+        _burn(msg.sender, msg.sender, amount, data, "");
+    }
+
+    /**
+    * @dev Burn the amount of tokens on behalf of the address from
+    * @param from address token holder address. Set to 0x0 to use msg.sender as token holder
+    * @param amount uint256 amount of tokens to transfer
+    * @param data bytes extra information provided by the token holder
+    * @param operatorData bytes extra information provided by the operator (if any)
+     */
+    function operatorBurn(address from, uint256 amount, bytes calldata data, bytes calldata operatorData) external {
+        address holder = from == address(0) ? msg.sender : from;
+        _burn(msg.sender, holder, amount, data, operatorData);
+    }
+
+    /**
+    * @return the name of the token.
+    */
+    function name() public view returns (string memory) {
+        return _name;
+    }
+
+    /**
+    * @return the symbol of the token.
+    */
+    function symbol() public view returns (string memory) {
+        return _symbol;
+    }
+
+    /**
+     * @dev Total number of tokens in existence
+     */
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    /**
+    * @dev Gets the balance of the specified address.
+    * @param tokenHolder The address to query the balance of.
+        * @return uint256 representing the amount owned by the specified address.
+     */
+    function balanceOf(address tokenHolder) public view returns (uint256) {
+        return _balances[tokenHolder];
+    }
+
+    /**
+    * @dev Gets the token's granularity,
+    * i.e. the smallest number of tokens (in the basic unit)
+    * which may be minted, sent or burned at any time
+    * @return uint256 granularity
+     */
+    function granularity() public view returns (uint256) {
+        return _granularity;
+    }
+
+    /**
+    * @dev Get the list of default operators as defined by the token contract.
+    * @return address[] default operators
+    */
+    function defaultOperators() public view returns (address[] memory) {
+        return _defaultOperatorsArray;
+    }
+
+    /**
+    * @dev Authorize an operator for the sender
+    * @param operator address to be authorized as operator
+     */
+    function authorizeOperator(address operator) public {
```

Comment:
I don't think we've done this in other contracts yet, but I want us to consider keeping this function `external` only, and having a separate `internal` function for extending the contract etc.

@nventuro What do you think?

Reply:
Yes - I wouldn't make the `internal` one just yet to keep this PR as simple as possible, but making these `external` sounds good.